lib/has_friendship/friendship.rb

```diff
@@ -1,16 +1,39 @@
 module HasFriendship
   class Friendship < ActiveRecord::Base
 
-    def self.check_one_side(friendable, friend)
-      find_by(friendable_id: friendable.id, friendable_type: friendable.class.base_class.name, friend_id: friend.id).present?
+    def self.relation_attributes(one, other)
+      {friendable_id: one.id, friendable_type: one.class.base_class.name, friend_id: other.id}
+    end
+
+    def self.create_relation(one, other, options)
+      relation = new relation_attributes(one, other)
+      relation.attributes = options
+      relation.save
+    end
+
+    def self.find_relation(friendable, friend)
+      where relation_attributes(friendable, friend)
+    end
+
+    def self.find_friend(friendable, friend)
+      find_relation(friendable, friend).where(status: 'accepted')
     end
 
     def self.exist?(friendable, friend)
-      check_one_side(friendable, friend) && check_one_side(friend, friendable)
+      find_relation(friendable, friend).any? && find_relation(friend, friendable).any?
+    end
+
+    # find not blocked frienships
+    def self.find_viable_friendship(friendable, friend)
+      find_relation(friendable, friend).where.not(status: "blocked").first
     end
 
-    def self.find_friendship(friendable, friend)
-      find_by(friendable_id: friendable.id, friendable_type: friendable.class.base_class.name, friend_id: friend.id)
+    singleton_class.send(:alias_method, :find_friendship, :find_viable_friendship )
```

Comment:
Yep, I think that would be better; I left it there because I did not want to change the design too much in case someone is using that method:)